server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -85,10 +93,55 @@ public Result dashboard(Http.Request request, Optional<String> search, Optional<
             pageInfo.getPageCount(),
             pageInfo.getPage(),
             search,
+            searchDate,
             request,
             messagesApi.preferred(request)));
   }
 
+  @Secure(authorizers = Authorizers.Labels.TI)
+  public Result updateDateOfBirth(Long tiId, Long applicantId, Http.Request request) {
+    Optional<CiviFormProfile> civiformProfile = profileUtils.currentUserProfile(request);
+
+    if (civiformProfile.isEmpty()) {
+      return unauthorized();
+    }
+    Optional<TrustedIntermediaryGroup> trustedIntermediaryGroup =
+        userRepository.getTrustedIntermediaryGroup(civiformProfile.get());
+    if (trustedIntermediaryGroup.isEmpty()) {
+      return notFound();
+    }
+    if (!trustedIntermediaryGroup.get().id.equals(tiId)) {
+      return unauthorized();
+    }
+    Form<UpdateApplicantDOB> form =
+        formFactory.form(UpdateApplicantDOB.class).bindFromRequest(request);
+    if (form.hasErrors()) {
+      return redirectToDashboardWithUpdateDateOfBirthError(form.errors().get(0).message(), form);
+    }
+    if (Strings.isNullOrEmpty(form.get().getDob())) {
+      return redirectToDashboardWithUpdateDateOfBirthError("Date Of Birth is required.", form);
+    }
+
+    try {
+      Applicant applicant =
+          userRepository.lookupApplicant(applicantId).toCompletableFuture().join().get();
+      applicant.getApplicantData().setDateOfBirth(form.get().getDob());
+      userRepository.updateApplicant(applicant);
+    } catch (DateOfBirthExistsException e) {
+      String trustedIntermediaryUrl = baseUrl + "/trustedIntermediaries";
+
+      return redirectToDashboardWithUpdateDateOfBirthError(
+          "Date Of birth for this applicant exists.  Cannot Update DOB. "
+              + " Direct applicant to sign in and go to"
+              + " "
+              + trustedIntermediaryUrl,
+          form);
+    }
+    return redirect(
+        routes.TrustedIntermediaryController.dashboard(
+            Optional.empty(), Optional.empty(), Optional.empty()));
```

Comment:
suggestion (blocking): add param name comments for each of these empty optionals `/* paramName= */ Optional.empty()`